test/abstractarray.jl

```diff
@@ -722,78 +746,89 @@ A = TSlowNIndexes(rand(2,2))
 @test A[1,1] == A.data[1]
 @test first(A) == A.data[1]
 
-#16381
-@inferred size(rand(3,2,1), 2, 1)
-@inferred size(rand(3,2,1), 2, 1, 3)
+@testset "#16381" begin
+    @inferred size(rand(3,2,1), 2, 1)
+    @inferred size(rand(3,2,1), 2, 1, 3)
 
-@test @inferred(indices(rand(3,2)))    == (1:3,1:2)
-@test @inferred(indices(rand(3,2,1)))  == (1:3,1:2,1:1)
-@test @inferred(indices(rand(3,2), 1)) == 1:3
-@test @inferred(indices(rand(3,2), 2)) == 1:2
-@test @inferred(indices(rand(3,2), 3)) == 1:1
+    @test @inferred(indices(rand(3,2)))    == (1:3,1:2)
+    @test @inferred(indices(rand(3,2,1)))  == (1:3,1:2,1:1)
+    @test @inferred(indices(rand(3,2), 1)) == 1:3
+    @test @inferred(indices(rand(3,2), 2)) == 1:2
+    @test @inferred(indices(rand(3,2), 3)) == 1:1
+end
 
-#17088
-let n = 10
+@testset "#17088" begin
+    n = 10
     M = rand(n, n)
-    # vector of vectors
-    v = [[M]; [M]] # using vcat
-    @test size(v) == (2,)
-    @test !issparse(v)
-    # matrix of vectors
-    m1 = [[M] [M]] # using hcat
-    m2 = [[M] [M];] # using hvcat
-    @test m1 == m2
-    @test size(m1) == (1,2)
-    @test !issparse(m1)
-    @test !issparse(m2)
-end
-
-# isinteger and isreal
-@test all(isinteger, Diagonal(rand(1:5, 5))) # reducing isinteger(...) deprecated
-@test isreal(Diagonal(rand(5)))
-
-# unary ops
-let A = Diagonal(rand(1:5,5))
+    @testset "vector of vectors" begin
+        v = [[M]; [M]] # using vcat
+        @test size(v) == (2,)
+        @test !issparse(v)
+    end
+    @testset "matrix of vectors" begin
+        m1 = [[M] [M]] # using hcat
+        m2 = [[M] [M];] # using hvcat
+        @test m1 == m2
+        @test size(m1) == (1,2)
+        @test !issparse(m1)
+        @test !issparse(m2)
+    end
+end
+
+@testset "isinteger and isreal" begin
+    @test isinteger(Diagonal(rand(1:5,5)))
+    @test isreal(Diagonal(rand(5)))
+end
+
+@testset "unary ops" begin
+    A = Diagonal(rand(1:5,5))
```

Comment:
Okay, I get it, so I should have let the `let` stay to be 100% sure that there could never be any behaviour changes in the file. If some tests depended on an `A` defined at the top, this would potentially change tests below the testset in question. should I change it back?